cmd/cosign/cli/sign/sign_blob.go

```diff
@@ -263,3 +270,31 @@ func extractCertificate(ctx context.Context, sv *SignerVerifier) ([]byte, error)
 	}
 	return nil, nil
 }
+
+func getHashFunction(sv *SignerVerifier) (crypto.Hash, error) {
+	pubKey, err := sv.PublicKey()
+	if err != nil {
+		return crypto.Hash(0), fmt.Errorf("error getting public key: %w", err)
+	}
+
+	// TODO: Ideally the SignerVerifier should have a method to get the hash function
+	// Cosign uses ED25519ph by default for ED25519 keys
+	algo, err := signature.GetDefaultAlgorithmDetails(pubKey, signatureoptions.WithED25519ph())
```

Comment:
this would be another place to condition the PH variant on if we're uploading to rekor